pkg/codegen/docs/gen.go

```diff
@@ -1789,6 +1791,22 @@ func (mod *modContext) genResource(r *schema.Resource) resourceDocArgs {
 		}
 	}
 
+	// Set a cap on how many auxiliary types the docs will include. We do that to limit the maximum size of
+	// a doc page. The default limit is rather high - we tried setting it to 200 for everyone but got
+	// pushback from third-party consumers who have legitimate use cases with more that 200 types.
+	// Therefore, we currently apply a smaller limit of 200 to packages that we know have some bloat in
+	// their types (AWS and AWS Native).
+	// See https://github.com/pulumi/pulumi/issues/15507#issuecomment-2064361317
+	//
+	// Schema Tools will print a warning every time a new resources gets an increase in number of types
+	// that is beyond 200. This should help us catch new instances and make decisions whether to include all
+	// types or add a limit. The default is including all types up to 1000.
+	maxNestedTypes := 1000
+	switch mod.pkg.Name() {
+	case "aws", "aws-native":
```

Comment:
This makes me wonder if we could potentially make this configurable at the provider/schema level instead of hardcoding numbers for different packages?  I don't know how exactly this codegen'ing works, but from an outside perspective it would make more sense if the provider had control over this, rather than having codegen hardcode provider names here.

Reply:
Yeh I suppose this could go in the schema Meta block?

Reply:
I had this idea too, but I would argue that this is not actually a provider's concern. It's a peculiarity of the way we display docs and a limitation of our current docs toolchain. I'm not sure I like exposing this low-value setting in the schema...

Reply:
It doesn't feel like codegen's concern to deal with specific packages either :-) Unless we're planning to fix this peculiarity at some point, I guess we'll have to live with it somehow, and having to add providers here whenever we need a different number feels not so great.  I'd be happy with having a default of 1000 here, and just have it be an optional setting in the schema.  

That said if that's too much work and we just need to fix this for now I'm not going to oppose this change too strongly, but I don't think it's the right way to go longer term.

Reply:
Is there a way to apply the lower limit to all pulumi 1st party packages and keep it high for all others?

Reply:
Not that I know of unfortunately.

Reply:
> I had this idea too, but I would argue that this is not actually a provider's concern. It's a peculiarity of the way we display docs and a limitation of our current docs toolchain.

Fair, I'm happy with this as is.

Reply:
Maybe? But as others are happy with this I think we can also go ahead with it as it is and the revisit in the future if we end up needing to hardcode more providers here.